uclibc dpes not seem to handle a leading ':' in LD_LIBRARY_PATH.
Eg. LD_LIBRARY_PATH=":/path"

--- ./buildtools/wafsamba/samba_utils.py.orig	2013-08-01 13:16:13.425790642 +0000
+++ ./buildtools/wafsamba/samba_utils.py	2013-08-01 13:17:52.063543307 +0000
@@ -54,10 +54,9 @@
 def ADD_LD_LIBRARY_PATH(path):
     '''add something to LD_LIBRARY_PATH'''
     if 'LD_LIBRARY_PATH' in os.environ:
-        oldpath = os.environ['LD_LIBRARY_PATH']
+        newpath = os.environ['LD_LIBRARY_PATH'].split(':')
     else:
-        oldpath = ''
-    newpath = oldpath.split(':')
+        newpath = []
     if not path in newpath:
         newpath.append(path)
         os.environ['LD_LIBRARY_PATH'] = ':'.join(newpath)
